doc/RichTextArea/RichTextArea.md

```diff
@@ -0,0 +1,295 @@
+# Rich Text Area Control
+
+Andy Goryachev
+
+<andy.goryachev@oracle.com>
+
+
+## Summary
+
+Introducing a RichTextArea control for displaying and editing of rich text.
+
+![rich text area screenshot](rich-text-area.png)
+
+
+
+## Goals
+
+**RichTextArea** control enables support for a number of common use cases:
+
+- read-only presentation of rich text information (help pages, documentation, etc.)
+- a simple editor similar to WordPad or TextEdit level, suitable for note taking or message editing.
+- a code editor with syntax highlighting
+- an editor which combines rich text with interactive content, such as a code notebook
+- enable extension and customization using via the input map
+
+
+
+## Non-Goals
+
+The following list represents features RichTextArea does not support:
+
+- "wide" models with long (10K+ symbols) paragraphs (for performance reasons)
+- applications requiring arbitrary text/graphics positioning (MS Word)
+- desktop publishing application that require precise control of text appearance (Adobe InDesign)
+- multiple or rectangular selection segments
+- embedded tables
+
+
+
+## Motivation
+
+For a long time, JavaFX has lacked a dedicated rich text area control, resulting in a functional gap in relation to Swing with its StyledEditorKit/JEditorPane.  
+
+The new RichTextArea control intends to bridge this gap by providing a dedicated control for displaying and editing rich text.
+
+The main design goal is to provide a good enough control to be useful out-of-the box, as well as open to extension by the application developers.
```

Comment:
Maybe change "a good enough control"  to "a control that is fully-featured enough" or "a control that is complete enough"